mcs/build/profiles/basic.make

```diff
@@ -118,8 +118,12 @@ endif
 
 $(PROFILE_EXE): $(topdir)/build/common/basic-profile-check.cs $(GENSOURCES_CS)
 	$(MAKE) $(MAKE_Q) -C $(topdir)/packages
-	$(BOOTSTRAP_MCS) /warn:0 /noconfig /langversion:latest /r:System.dll /r:mscorlib.dll /out:$@ $<
-	$(BOOTSTRAP_MCS) /noconfig /langversion:latest /r:mscorlib.dll /r:System.dll /r:System.Core.dll /out:$(GENSOURCES_EXE) $(GENSOURCES_CS)
+	$(BOOTSTRAP_MCS) /warn:0 /noconfig /langversion:latest /r:System.dll /r:mscorlib.dll /out:$@.tmp $<
+	$(BOOTSTRAP_MCS) /noconfig /langversion:latest /r:mscorlib.dll /r:System.dll /r:System.Core.dll /out:$(GENSOURCES_EXE).tmp $(GENSOURCES_CS)
+	- rm $(GENSOURCES_EXE)
+	mv $(GENSOURCES_EXE).tmp $(GENSOURCES_EXE)
+	- rm $@
```

Comment:
Why not to use `mv -f` ?

Reply:
wasn't familiar with it, but I suspect it's not atomic so it would reintroduce the race (at a lower chance of happening)

Reply:
Are you saying `mv -f` is not atomic but `rm + mv` is atomic?

Reply:
Only a single mv is atomic, so I'm not certain whether ```mv -f```'s observed behavior will do what I want. I can do more reading about it if you want but the manpage did not seem to make any guarantees - this isn't something googleable but maybe i can read through the source to gnu mv

Reply:
It's not worth it, let's get this in

Reply:
I read through the source code and it looks like -f is just implemented as rm + mv so I'll switch to that.